kanidmd/src/lib/core/mod.rs

```diff
@@ -382,6 +382,20 @@ pub fn domain_rename_core(config: &Configuration, new_domain_name: &str) {
         }
     };
 
+    // make sure we're actually changing the domain name...
+    match task::block_on(qs.read_async()).get_domain_name() {
+        Ok(old_domain_name) => {
+            if &old_domain_name == &new_domain_name {
+                admin_info!("Domain name not changing, stopping.");
+                std::process::exit(125); // ECANCELED
```

Comment:
I'm not sure about this exit here, just because we are in a transaction still at this point, and the return would at least cause that to be ended. Do we have an especial need to return a different exit code? I think for the purpose of this, if we are "changi" the name to the same name, it's still a "success" anyway ....

Reply:
Yeah, it's a little confusing to look at, but the qs.read_async() starts a txn, and it lasts for the duration of this block while you get the domain name.